server/schedule/operator/create_operator.go

```diff
@@ -34,6 +34,9 @@ func CreateAddPeerOperator(desc string, cluster opt.Cluster, region *core.Region
 
 // CreatePromoteLearnerOperator creates an operator that promotes a learner.
 func CreatePromoteLearnerOperator(desc string, cluster opt.Cluster, region *core.RegionInfo, peer *metapb.Peer) (*Operator, error) {
+	if region.GetPendingPeer(peer.GetId()) != nil {
```

Comment:
This check is more reasonable in the builder.

Reply:
+1

Reply:
Updated.